### ISSUE TITLE
Enhancement: prepare for repo release 2.1 (now release 2.3)

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/mkiocccentry#1038]
### Is there an existing issue for this?

- [X] I have searched for existing issues and did not find anything like this

### Describe the enhancement

Given issue # 1037 and a few other issues, we need to release an update to this repo.

### Relevant images, screenshots or other files

n/a

### Relevant links

n/a

### Anything else?

**SIGH** 😞

The purpose of this issue is to determine which bugs need to be fixed by the next release.

Here is the (growing(?)) list:

# TODO

[ ] issue #1037 
[x] issue #1036 
[x] issue #1035 
[x] issue #1033 
[x] Determine the answer to [GH-issuecomment-2563036384](https://github.com/ioccc-src/mkiocccentry/issues/1039#issuecomment-2563036384)

Note: the answer to the above comment appears to be to make two options of used together an error. To document this in main pages as well as the usage message of `jstrdecode(1)`. 

[x] issue #1039 in light of [GH-issuecomment-2563085398](https://github.com/ioccc-src/mkiocccentry/issues/1039#issuecomment-2563085398)

**Comment by lcn2** [comment]
Regarding [GH-issuecomment-2561259252](https://github.com/ioccc-src/temp-test-ioccc/issues/5#issuecomment-2561259252), please open up a new issue to discuss this problem, @xexyl.

**Comment by lcn2** [comment]
For issues that we determine need to be folded into this release, we need to add them to the TODO checklist of the top comment **AND** remove the "_((POST-IOCCC28))_" label from them as required.

**Comment by lcn2** [comment]
We recommend the you, @xexyl, issue separate PRs for each issue, if possible.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by lcn2** [comment]
# IMPORTANT NOTE ON PRIORITIES

BTW @xexyl 

In terms of priorities, working on getting ready for the **Great Fork Merge** such as noted in is of MUCH HIGHER priority.

- [GH-issuecomment-2561512948](https://github.com/ioccc-src/temp-test-ioccc/issues/2239#issuecomment-2561512948) 
- [GH-issuecomment-2561506920](https://github.com/ioccc-src/temp-test-ioccc/issues/2239#issuecomment-2561506920)

We can enter into IOCCC pending state without a repo 2.1 release.  We **CANNOT** enter into IOCCC pending state unless the **Great Fork Merge** is complete.  Lets be sure w can enter into IOCCC pending state **BEFORE THE END OF 2024**!

**Comment by xexyl** [comment]
I am a bit confused about the current release you say given that I am up to date. However as you say to open a new issue I will do that. I hope to be able to do that in a bit and hopefully be able to reply to your other comments today. If not I certainly will tomorrow morning. I have to do some other things and then I will see if I can start these things including the excellent news about the mailing list!

**Comment by xexyl** [comment]
As for the -15 stuff: that's actually not here either. I will try explaining that in more depth later. I know you were focusing on the server but it is not in the website scripts at all. I will explain in the new issue you asked me to open. I hope that I can get to that in about half an hour to an hour from now. Just have to do usual morning stuff. 

Merry Christmas 🎄🎁! (I have something else for that too .. later)

**Comment by xexyl** [comment]
I'll reply to this comment first. Then I'll open the new issue like you asked me to (or maybe check the mailing list email before I forget).

> Regarding [GH-issuecomment-2561259252](https://github.com/ioccc-src/temp-test-ioccc/issues/5#issuecomment-2561259252), please open up a new issue to discuss this problem, @xexyl.
> 

Will do .. hopefully today. I aim to do it this morning anyway.

> ## UPDATE 0
> > You suggested that I clone version 2.0. One problem is that the actual last release is 2.0.1. This fixed a critical compilation error in linux. That means that if someone clones/checks out the 2.0 release (which would seem odd to suggest this to others) they will not be able to compile jparse (in full) in linux. But having 2.0.1 introduced the issue of a missing variable argument (and still not sure how that slipped through).
> 
> Perhaps you misunderstood what we wrote in [GH-issuecomment-2560064926)](https://github.com/ioccc-src/temp-test-ioccc/issues/5#issuecomment-2560064926)?
> 
> We suggested that you, @xexyl, return to the **the official mkiocccentry repository 2.0.0 release**. Here is the ** mkiocccentry repository**:
> 
> https://github.com/ioccc-src/mkiocccentry/releases/tag/v2.0
> 
> The [mkiocccentry repository 2.0.0 release](https://github.com/ioccc-src/mkiocccentry/releases/tag/v2.0) has these `-V` outputs:
> 
> * chkentry -V
> 
> ```
> chkentry version: 1.0.2 2024-08-26
> jparse UTF-8 version: 2.0.5 2024-11-14
> jparse library version: 2.1.1 2024-11-17
> ```
> 
>  * mkiocccentry -V
> 
> ```
> mkiocccentry version: 1.0.8 2024-08-23
> jparse UTF-8 version: 2.0.5 2024-11-14
> jparse library version: 2.1.1 2024-11-17
> ```
> 
>  * txzchk -V
> 
> ```
> txzchk version: 1.0.8 2024-07-11
> jparse UTF-8 version: 2.0.5 2024-11-14
> jparse library version: 2.1.1 2024-11-17
> ```
> 
>  * jparse -V
> 
> ```
> jparse version: 1.2.3 2024-11-13
> jparse UTF-8 version: 2.0.5 2024-11-14
> jparse library version: 2.1.1 2024-11-17
> ```
> 
>  * jstrdecode -V
> 
> ```
> jstrdecode version: 2.1.2 2024-11-17
> jparse UTF-8 version: 2.0.5 2024-11-14
> jparse library version: 2.1.1 2024-11-17
> ```
> 
>  * jstrencode -V
> 
> ```
> jstrencode version: 2.1.1 2024-11-15
> jparse UTF-8 version: 2.0.5 2024-11-14
> jparse library version: 2.1.1 2024-11-17
> ```
> 

I understand this but my point is if we return to 2.0.0 then it means that others will have to do the same rather than just do a git clone. That I think would be a problem and it also means the FAQ and maybe guidelines have to be updated.

> This code is the **exact same code** as found at the [top of the master branch](https://github.com/ioccc-src/mkiocccentry/commit/5a9dc8e446f0f0c2ddb7af9d95e5ba54344f6919).
> 
> This code is the **exact same code** as found in the [Source code (tar.gz)](https://github.com/ioccc-src/mkiocccentry/archive/refs/tags/v2.0.tar.gz) compressed tarball.
> 
> This code compiles under Linux without any problems.

Perhaps it's only in some linux distributions? Because in rocky linux it failed.

> 
> This code, when `make clobber all install`-ed under macOS 15.2 allows `make www` to run without error nor adverse modification to generated HTML files.

That's true that that version does. But since it doesn't compile in all linux distributions ... or at least jparse itself did not, and since 2.0.0 is not the latest ... it seems problematic to ask users to get that release instead of the most recent one?

> 
> ## UPDATE 1
> We had someone else who has a mac, clone both [this repo](https://github.com/ioccc-src/temp-test-ioccc) and the [mkiocccentry repo](https://github.com/ioccc-src/mkiocccentry).
> 

I'm unsure how they could have done that with the most recent release since the version of jstrdecode(1) in the most recent release validates JSON first and it will then cause an error because in the `make www` it removes the quotes from the JSON. I'll mention more in the new issue and then will give more details.

> They did (after installing the pre-requite external tools such as git, jparse, etc.)
> 
> ```shell
> git clone https://github.com/ioccc-src/mkiocccentry.git
> cd mkiocccentry
> make clobber all test
> sudo make install
> 
> cd ..
> git clone https://github.com/ioccc-src/temp-test-ioccc.git
> cd temp-test-ioccc
> make www
> ```
> 
> They had no errors, nor any of that " - 15 " stuff mention in [GH-issuecomment-2558553519](https://github.com/ioccc-src/temp-test-ioccc/issues/5#issuecomment-2558553519), etc.
> 

Okay but this is strange.

As for the `- 15` stuff that's not without adding the necessary option to `jstrdecode` in the website scripts to prevent validation of JSON so that it does not result in an error. I'll give an example in the new issue. So it's expected they would not have it; what's not expected is for them to be able to run those commands and be able to do `make www`.

> We are **NOT** seeing any problems when doing `make www` or when doing `make quick_www` that you seem to report in regards to `jstrdecode(1)`.
> 

But since you have a different version from `mkiocccentry(1)` most recent release this makes sense. Because the request to validate JSON came in the following version.

> ### Is your environment the problem?
> In regards to your reported `make www` or `make quick_www` problems as well as that " - 15 " stuff mention in [GH-issuecomment-2558553519](https://github.com/ioccc-src/temp-test-ioccc/issues/5#issuecomment-2558553519) stuff"

Well unfortunately my fedora box is still down and rocky linux has problems with the most recent pandoc: though maybe I can try the GitHub repo. However as it's the server I do try and limit the number of things installed.

I don't think it's my environment. However I can try a `make install` from the mkiocccentry repo and see. If it works something will be funny.

> 
> Given the above, we really wonder of you don't have some corrupted or modified or different code than what is available under the [mkiocccentry repository 2.0.0 release](https://github.com/ioccc-src/mkiocccentry/releases/tag/v2.0) code base.
> 
> Perhaps you have not done a:
> 
> ```shell
> git clone https://github.com/ioccc-src/mkiocccentry.git
> cd mkiocccentry
> sudo make uninstall
> make clobber all test
> sudo make install
> 
> cd ..
> git clone https://github.com/ioccc-src/temp-test-ioccc.git
> cd temp-test-ioccc
> make www
> ```
> 
> Or perhaps your "$PATH" is favoring your "jparse code base", or your LD_LIBRARY_PATH cases you not to use the libs installed by the cloned mkiocccentry, or?

> 
> ## UPDATE 2
> ### Fixing jstrdecode for repo release 2.1?
> Yes, there may be something wrong with the `jstrdecode(1)` has you stated, but it is worth the risk of updating the code prior to the propose repo release 2.1? Perhaps yes.

I think it is because there is a memory error in `-h` due to missing fprintf arg :( People who look at it will likely report it. Since this bug is already fixed it shouldn't be a problem. However since there is something else to consider I'll wait until I open the new issue.


> 
> Perhaps there is a case for such a mod? If so we request that you push PRs for the issues identified [TODO list of the top comment of GH-issue-2758345687](https://github.com/ioccc-src/mkiocccentry/issues/1038#issue-2758345687) **BEFORE** addressing the `jstrdecode(1)` issue. That way we can evaluate the the `jstrdecode(1)` change you request and if there is a problem, tag as release 2.1, the commit that resolves the [TODO list of the top comment of GH-issue-2758345687](https://github.com/ioccc-src/mkiocccentry/issues/1038#issue-2758345687).


Hmm. Please elaborate here. If I sync the bug fix (`-h` option) that will also have the newer feature that you needed in `jstrdecode(1)` which will make (or so it seems on my end) `make www` fail due to validating the JSON. Now I did put an option to disable that but enabling that in the website scripts is what made the `- 15` show up (which is really strange).

I'll look at the mailing list now and then open that new issue.

I have some time now but later on in the day I will probably be unable to do more. Tomorrow should be fine though!

**Comment by xexyl** [comment]
# GREAT NEWS: it seems it was a false alarm, the jstrdecode(1) in this repo, due to environment!

It WAS my environment! This is so bizarre .. esp as it also compiles in rocky linux! I'm so sorry about all the comments that appear to be, at this time, false alarms! See below.

Indeed with the most recent releasee it actually does work! The problem is I must have made a commit to fix the compilation error that I noticed in rocky linux but I did not push it as the code freeze happens. I did the pull but for whatever reason `git status` says it's up to date. That's why I thought there was an issue, and that the release here was actually 2.0.1: even though it seemed odd. There's another really odd thing. But first...

With the most recent release of mkiocccentry outside my fork the version is correct!

This explains everything now. So the bug with `-h` is not there at all! That was only in the commit I had apparently made which has been fixed already in jparse! And the validation of JSON by default here is not there as the version of `jstrdecode1)` in mkiocccentry is the version before that!

So there is no problem here.

That being said: due to an issue you reported in the website wrt jstrdecode the tool now (2.1.3 I think, one version higher than the version here anyway) now validates JSON first. That means the website scripts are broken because the quotes are removed and then the JSON is invalid!

Adding the option to skip validation in that tool is what caused the `- 15` to appear: something I cannot understand at all.

So for now the solution for me to work on the website, if it involves JSON, is to use the mkiocccentry 2.0.0.

The question is whether the default behaviour of jstrdecode should validate JSON or if using `-j` (currently that disables validation) should make it validate it. It did solve a problem but I must admit it's been annoying to have to always include `"` in input so maybe it's better to make `-j` enable it rather than disable it? Perhaps so?

This question can be determined later but it does seem that at least the bug I reported is not here.

What is **EXTREMELY** odd is that this all compiles in rocky linux when there was a problem earlier (at least in jparse)!

**Comment by xexyl** [comment]
> # GREAT NEWS: it seems it was a false alarm, the jstrdecode(1) in this repo, due to environment!
> It WAS my environment! This is so bizarre .. esp as it also compiles in rocky linux! I'm so sorry about all the comments that appear to be, at this time, false alarms! See below.
> 
> Indeed with the most recent releasee it actually does work! The problem is I must have made a commit to fix the compilation error that I noticed in rocky linux but I did not push it as the code freeze happens. I did the pull but for whatever reason `git status` says it's up to date. That's why I thought there was an issue, and that the release here was actually 2.0.1: even though it seemed odd. There's another really odd thing. But first...
> 
> With the most recent release of mkiocccentry outside my fork the version is correct!
> 
> This explains everything now. So the bug with `-h` is not there at all! That was only in the commit I had apparently made which has been fixed already in jparse! And the validation of JSON by default here is not there as the version of `jstrdecode1)` in mkiocccentry is the version before that!
> 
> So there is no problem here.
> 
> That being said: due to an issue you reported in the website wrt jstrdecode the tool now (2.1.3 I think, one version higher than the version here anyway) now validates JSON first. That means the website scripts are broken because the quotes are removed and then the JSON is invalid!
> 
> Adding the option to skip validation in that tool is what caused the `- 15` to appear: something I cannot understand at all.
> 
> So for now the solution for me to work on the website, if it involves JSON, is to use the mkiocccentry 2.0.0.
> 
> The question is whether the default behaviour of jstrdecode should validate JSON or if using `-j` (currently that disables validation) should make it validate it. It did solve a problem but I must admit it's been annoying to have to always include `"` in input so maybe it's better to make `-j` enable it rather than disable it? Perhaps so?
> 
> This question can be determined later but it does seem that at least the bug I reported is not here.
> 
> What is **EXTREMELY** odd is that this all compiles in rocky linux when there was a problem earlier (at least in jparse)!

# QUESTION

Do you wish me to open an issue about jstrdecode default behaviour wrt to validating JSON? If so should it be in jparse, the website or here? I guess the jparse repo might be the best place but I don't know. I do think it might be an issue to have it by default validate it ... but of course it does solve a problem: though a problem I no longer remember the details.

**Comment by lcn2** [comment]
Glad to hear the **GOOD NEWS** continued in [GH-issuecomment-2561929009](https://github.com/ioccc-src/mkiocccentry/issues/1038#issuecomment-2561929009)

## Regarding repo version numbers

Please be aware that the current and latest **repo release version** is "**2.0 2024-11-17**":

[mkiocccenrtry repo version 2.0 2024-11-17](https://github.com/ioccc-src/mkiocccentry/releases/tag/v2.0)

The repo release version is **NOT** the version of the individual tools or libraries.

This issue seeks to advance the **repo release version** from "**2.0 2024-11-17**" to some form of "**2.1 YYYY-MM-DD**".   This will involve resolving issues noted in the top level TODO comment.  That in turn will likely involve changing the version number of individual tools or libraries.

**Comment by xexyl** [comment]
> Glad to hear the **GOOD NEWS** continued in [GH-issuecomment-2561929009](https://github.com/ioccc-src/mkiocccentry/issues/1038#issuecomment-2561929009)
> 
> 
> 
> ## Regarding repo version numbers
> 
> 
> 
> Please be aware that the current and latest **repo release version** is "**2.0 2024-11-17**":
> 
> 
> 
> [mkiocccenrtry repo version 2.0 2024-11-17](https://github.com/ioccc-src/mkiocccentry/releases/tag/v2.0)
> 
> 
> 
> The repo release version is **NOT** the version of the individual tools or libraries.
> 
> 
> 
> This issue seeks to advance the **repo release version** from "**2.0 2024-11-17**" to some form of "**2.1 YYYY-MM-DD**".   This will involve resolving issues noted in the top level TODO comment.  That in turn will likely involve changing the version number of individual tools or libraries.
> 
> 

Of course. I see that once I looked at another clone of the repo. Not sure how it even happened. 

But at least it works now. Thanks.

**Comment by lcn2** [comment]
> # QUESTION
> Do you wish me to open an issue about jstrdecode default behaviour wrt to validating JSON? If so should it be in jparse, the website or here? I guess the jparse repo might be the best place but I don't know. I do think it might be an issue to have it by default validate it ... but of course it does solve a problem: though a problem I no longer remember the details.

We suggest that you, @xexyl, open a new issue to address that matter.  In that new issue we can discuss of such a change should be applied before or after the repo version 2.1 release.

**Comment by lcn2** [comment]
> # QUESTION
> Do you wish me to address the issues you opened and make a pull request **HERE**? If you do I can do that: though probably not until tomorrow.

We don't understand your question.

**Comment by xexyl** [comment]
> > # QUESTION
> > Do you wish me to address the issues you opened and make a pull request **HERE**? If you do I can do that: though probably not until tomorrow.
> 
> We don't understand your question.

This is already answered. It was to do with the new issues. I wanted to know if you wanted me to open new pull requests. But I'll be doing that later today so all is good.

**Comment by xexyl** [comment]
> > # QUESTION
> > Do you wish me to open an issue about jstrdecode default behaviour wrt to validating JSON? If so should it be in jparse, the website or here? I guess the jparse repo might be the best place but I don't know. I do think it might be an issue to have it by default validate it ... but of course it does solve a problem: though a problem I no longer remember the details.
> 
> We suggest that you, @xexyl, open a new issue to address that matter. In that new issue we can discuss of such a change should be applied before or after the repo version 2.1 release.

Will do this.

**Comment by xexyl** [comment]
Issue opened .. and various issues fixed. Running `make release` now. I'm unfortunately having a real problem focusing my eyes so it's all I can do now. I hope I can continue more later today. I know there's a lot to do on the website front too though ... more than I wish there was. Trying to finish this before the end of the year but it's fast approaching :(

**Comment by xexyl** [comment]
Oh great ... pull request link not showing up and I don't know what to do. I fixed three issues too :(

**Comment by lcn2** [comment]
For now, we have dropped issue #1034 from the TODO list

**Comment by xexyl** [comment]
> For now, we have dropped issue #1034 from the TODO list

Sounds like a good idea!

**Comment by lcn2** [comment]
So we have one item in to do list, issue #1037 

However we added another TODO:

- Determine the answer to [GH-issuecomment-2563036384](https://github.com/ioccc-src/mkiocccentry/issues/1039#issuecomment-2563036384)

Answering that question might be a good idea to do before release 2.1 of this repo, as if it is resolved in a simple way, the fix could be folded into this proposed 2.1 repo release. 

Thus, we have added that question as a to-do for this issue.

**Comment by xexyl** [comment]
> # Status update
> So we have one item in to do list, issue #1037
> 
> However we added another TODO:
> 
> * Determine the answer to [GH-issuecomment-2563036384](https://github.com/ioccc-src/mkiocccentry/issues/1039#issuecomment-2563036384)
> 
> Answering that question might be a good idea to do before release 2.1 of this repo, as if it is resolved in a simple way, the fix could be folded into this proposed 2.1 repo release.
> 
> Thus, we have added that question as a to-do for this issue.

I answered it and yes the fix is simple. Just have to invert the meaning of `-j`, update the version, update the man page and documentation and then sync it over. Though as you know the Makefile.local system was removed. I was going to add it back but then I thought I'd go for the moving the files back and forth and that proved a problem. But since we're not going to do that here maybe it's okay there too and I could more easily add it back (I have the old bug_report.sh script that checks it and it would just be a few other changes.).

Let me know what you wish.

**Comment by lcn2** [comment]
> > # Status update
> 
> > So we have one item in to do list, issue #1037
> 
> > 
> 
> > However we added another TODO:
> 
> > 
> 
> > * Determine the answer to [GH-issuecomment-2563036384](https://github.com/ioccc-src/mkiocccentry/issues/1039#issuecomment-2563036384)
> 
> > 
> 
> > Answering that question might be a good idea to do before release 2.1 of this repo, as if it is resolved in a simple way, the fix could be folded into this proposed 2.1 repo release.
> 
> > 
> 
> > Thus, we have added that question as a to-do for this issue.
> 
> 
> 
> I answered it and yes the fix is simple. Just have to invert the meaning of `-j`, update the version, update the man page and documentation and then sync it over. Though as you know the Makefile.local system was removed. I was going to add it back but then I thought I'd go for the moving the files back and forth and that proved a problem. But since we're not going to do that here maybe it's okay there too and I could more easily add it back (I have the old bug_report.sh script that checks it and it would just be a few other changes.).
> 
> 
> 
> Let me know what you wish.

Please test the above proposed default behavior change against the "other repo" while doing a full `make www`.  

If all is well and there are no changes to the HTML content generated by `make www` in the "other repo", then please PR to resolve issue #1037.

**Comment by xexyl** [comment]
> > > # Status update
> > 
> > 
> > > So we have one item in to do list, issue #1037
> > 
> > 
> > > 
> > 
> > 
> > > However we added another TODO:
> > 
> > 
> > > 
> > 
> > 
> > > * Determine the answer to [GH-issuecomment-2563036384](https://github.com/ioccc-src/mkiocccentry/issues/1039#issuecomment-2563036384)
> > 
> > 
> > > 
> > 
> > 
> > > Answering that question might be a good idea to do before release 2.1 of this repo, as if it is resolved in a simple way, the fix could be folded into this proposed 2.1 repo release.
> > 
> > 
> > > 
> > 
> > 
> > > Thus, we have added that question as a to-do for this issue.
> > 
> > 
> > I answered it and yes the fix is simple. Just have to invert the meaning of `-j`, update the version, update the man page and documentation and then sync it over. Though as you know the Makefile.local system was removed. I was going to add it back but then I thought I'd go for the moving the files back and forth and that proved a problem. But since we're not going to do that here maybe it's okay there too and I could more easily add it back (I have the old bug_report.sh script that checks it and it would just be a few other changes.).
> > Let me know what you wish.
> 
> Please test the above proposed default behavior change against the "other repo" while doing a full `make www`.
> 
> If all is well and there are no changes to the HTML content generated by `make www` in the "other repo", then please PR to resolve issue #1037.

I can run `make www` after running `make install` from jparse, sure. I don't know if I can finish it today, however, but I'll at least start it. If I have to start over tomorrow morning that's okay. But code is updates are done anyway.

**Comment by xexyl** [comment]
Hmm .. wait. Something is not quite right. Or maybe I discovered why we had the old way? Checking this.

**Comment by xexyl** [comment]
Oh no. That's not true. It's fine now. Okay I can start make www but it might not be done until tomorrow.

**Comment by xexyl** [comment]
Actually .. I am not going to do it today. I'll do it tomorrow morning. I won't be here much longer I'm afraid.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by lcn2** [comment]
Unless you think it would be a mistake, we have added to the above TODO list:

-  issue #1039 in light of [GH-issuecomment-2563085398](https://github.com/ioccc-src/mkiocccentry/issues/1039#issuecomment-2563085398)

**Comment by lcn2** [comment]
We plan to create a new release 2.2 issue, add links to the open issues as its TODO list (like we did for this one), and then close this one as "resolved".

**Comment by xexyl** [comment]
I believe that this is resolved with the commits I made earlier today? (And the other issue is resolved in that case.)

**Comment by lcn2** [comment]
> I believe that this is resolved with the commits I made earlier today? (And the other issue is resolved in that case.)

Very close, @xexyl.  See our suggestion in [GH-issuecomment-2566672002](https://github.com/ioccc-src/mkiocccentry/pull/1050#issuecomment-2566672002).  With that we think we can release a new mkiocccentry repo.

BTW: We suggest that you call that repo version 2.2.

**Comment by lcn2** [comment]
BTW: If we do a `make all.update_from_clone`, @xexyl, the some of your recent code improvements to this repo would revert.

You might wish to update "the other repo" (jparse).

**Comment by xexyl** [comment]
> BTW: If we do a `make all.update_from_clone`, @xexyl, the some of your recent code improvements to this repo would revert.
> 
> You might wish to update "the other repo" (jparse).

That's strange .. I did that exactly. I'll look at it.

**Comment by xexyl** [comment]
Just tried `make jparse.recreate_clone jparse.update_from_clone` and no changes.

**Comment by lcn2** [comment]
> Just tried `make jparse.recreate_clone jparse.update_from_clone` and no changes.

We are testing things now ... so far so good ...

**Comment by lcn2** [comment]
With the [repo version 2.3 release](https://github.com/ioccc-src/mkiocccentry/releases/tag/v2.3) this issue is now closed.
